.goreleaser.yaml

```diff
@@ -0,0 +1,129 @@
+# run a local "snapshot" build without publishing:
+#    TRAVIS_BUILD_NUMBER=1 goreleaser --snapshot --rm-dist --debug
+#
+project_name: chezmoi
+
+builds:
+- binary: chezmoi
+  env:
+    - CGO_ENABLED=0
+  ldflags:
+     - -s -w
+  goos:
+    - linux
+    - darwin
+    # TODO: uncomment when compatible with windows. currently fails on 'undefined: syscall.Stat_t'
+    # - windows
+    - freebsd
+    - openbsd
+    # - dragonfly
+    # - netbsd
+  goarch:
+    - 386
+    - amd64
+    - arm
+    - arm64
+  goarm:
+    - ''
+  ignore:
+    - goos: darwin
+      goarch: 386
+    - goos: windows
+      goarch: 386
+
+## generate .tar.gz and .zip archives
+archive:
+  replacements:
+    386: i386
+  format_overrides:
+  - goos: windows
+    format: zip
+
+checksum:
+  name_template: 'checksums.txt'
+
+snapshot:
+  name_template: "{{ .Tag }}-next"
+
+changelog:
+  sort: asc
+  filters:
+    exclude:
+    - '^docs:'
+    - '^test:'
+    - Merge pull request
+    - Merge branch
+
+## generate RPM and DEB packages
+nfpm:
+  vendor: "TODO"
+  homepage: "https://github.com/twpayne/chezmoi"
+  maintainer: Tom Payne <TODO@TODO.tld>
+  description: "chezmoi is a tool for managing your dotfiles across multiple machines."
+  license: MIT
+  formats:
+    - deb
+    - rpm
+  overrides:
+    rpm:
+      name_template: "{{ .ProjectName }}-{{ .Version }}-{{ .Env.TRAVIS_BUILD_NUMBER }}-{{ .Arch }}"
+      replacements:
+        amd64: x86_64
+        386: i686
+        arm: armhfp
+        arm64: aarch64
+    deb:
+      name_template: "{{ .ProjectName }}_{{ .Version }}-{{ .Env.TRAVIS_BUILD_NUMBER }}_{{ .Os }}_{{ .Arch }}"
+      replacements:
+        386: i386
+        arm: armel
+
+## generate a homebrew formula and push to github.com/twpayne/homebrew-taps
```

Comment:
If you want to host your own homebrew tap, do the following:

1. Create a new github repo `twpayne/homebrew-taps`
2. Uncomment this `brew:` block
3. Commit to master
4. create and push a new tag to github

When Goreleaser runs it will create a new file `chezmoi.rb` in the `twpayne/homebrew-taps` repo under the `Formula` directory. This allows macOS users to install with:

```console
brew install twpayne/taps/chezmoi
```

The kubefwd project has a good example of using goreleaser to manage a homebrew tap:

- https://github.com/txn2/kubefwd/blob/master/goreleaser.yml
- https://github.com/txn2/homebrew-tap/blob/master/Formula/kubefwd.rb

The homebrew formulas generated by goreleaser are not compatible with "homebrew-core", BTW. So if you intend this app to be part of the core homebrew package list (which it should be!) then this won't be necessary but it's great to have in the meantime.